keras_cv/models/segmentation/__internal__/segmentation_head.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class SegmentationHead(layers.Layer):
+    """Prediction head for the segmentation model
+
+    The head will take the output from decoder (eg FPN or ASPP), and produce production
+    (pixel level classifications) as the output for the model.
+
+    Args:
+        classes: int, the number of output classes for the prediction.
```

Comment:
Can we help users by clarifying whether this will include background class? It's likely we require 1+num_foreground_classes here, but user pass num_foreground_classes?
(assuming we will make this as public API)

Reply:
Good point - if this is only to be used in segmentation asking for foreground classes is a nice way to keep it clear what the user is to pass.  Another approach would be to call this output_dim instead of classes

Reply:
Done.